src/shogun/machine/IterativeMachine.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVEMACHINE_H__
+#define _ITERATIVEMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+
+#define SG_ADD8(param, name, description, ms_available)                        \
+	{                                                                          \
+		this->m_parameters->add(param, name, description);                     \
+		this->watch_param(                                                     \
+		    name, param,                                                       \
+		    AnyParameterProperties(                                            \
+		        description, ms_available, GRADIENT_NOT_AVAILABLE));           \
+		if (ms_available)                                                      \
+			this->m_model_selection_parameters->add(param, name, description); \
+	}
+
+	/** @brief Class IterativeLinearMachine implements an iterative model
+	 * whose training can be prematurely stopped, and in particular
+	 * resumed, anytime.
+	 */
+	template <class T>
+	class CIterativeMachine : public T
+	{
+	public:
+		/** Default constructor */
+		CIterativeMachine() : T()
```

Comment:
What is cool is that we avoid the usual mixin factory macro mess since we only have a default constructor anyways